breakout_3/breakout_3_answers.Rmd

```diff
@@ -1,77 +1,114 @@
+  ```
+  <STUDYID>-<SITEID>-<SUBJID>
+  ```
 
-3) Derive the age group(AGEGR1) into the following groups: <65, 65-80, >80
+2)  Derive the safety population flag (SAFFL) as "N" for the Screen Failures, and "Y" for all others
 
+3)  Derive the age group (AGEGR1) variable using the following groups:
 
-#### Prepare USUBJID
+-   \<65
+-   65-80
+-   \>80
 
-We already have USUBJID but we need it in a new format. Below use the `dplyr::mutate` function to update USUBJDID to this format <*STUDYID*>-<*SITEID*>-<*SUBJID*>. **Hint: The `paste0` function can be used to concatenate strings.**
+## Prepare USUBJID
+
+We already have USUBJID in our data but, we for this example we're going to put it in a new format.
+
+**Use the `mutate()` function from the `{dplyr}` package to update USUBJID in this format:**
+
+    <*STUDYID*>-<*SITEID*>-<*SUBJID*>. 
+
+*Hint: The `paste0` function can be used to concatenate strings.*
 
 ```{r prep_usubjid}
-dm <- dm %>%
-  mutate(USUBJID = paste0(STUDYID, "-", SITEID, "-", SUBJID))
+raw_dm %>%
+  mutate(USUBJID = paste0(STUDYID, "-", SITEID, "-", SUBJID)) %>% 
+  select(STUDYID, SITEID, SUBJID, USUBJID) %>% 
+  # These calls to `head()` and `kable()` are just to display the data nicely
+  head() %>% 
+  kable()
 ```
 
-#### Prepare SAFFL
+## Prepare SAFFL
 
-We can use the same `mutate` function to create the SAFFL variable. SAFFL should be "N" if the participant was screened out, and "Y" otherwise. **Hint: The `if_else` function can be used in a `mutate`.
+We can use the same `mutate()` function to create the SAFFL variable. SAFFL should be "N" if the participant was a screen failure, and "Y" otherwise.
+
+**Derive SAFFL using the logic described above.**
+
+*Hint: The `if_else()` function can be used in a `mutate()`.*
 
 ```{r prep_saffl}
-dm <- dm %>%
+raw_dm %>%
   mutate(
-    SAFFL = if_else(ARM == "Screen Failure",
-                    "N",
-                    "Y")
-  )
+    SAFFL = if_else(ARM != "Screen Failure",
+                    "Y",
+                    "N")
+  ) %>% 
+  select(USUBJID, ARM, SAFFL) %>% 
+  # These calls to `head()` and `kable()` are just to display the data nicely
+  head(10) %>% 
+  kable()
 ```
 
-#### Prepare AGEGR1
+## Prepare AGEGR1
+
+This is similar to the SAFFL logic except that there are multiple conditions for which we must test. The AGEGR1 variable will group the age of participants into less than 65 (\<65), 65 to 80 (65-80), and greater than 80 (\>80).
+
+**Derive AGEGR1 using the logic described above.**
 
-This is similar to the SAFFL logic except there are multiple conditions we must test for. The AGEGR1 variable will group the age of participants into less than 65(<65), 65 to 80(65-80), and greater than 80(>80). **Hint: Nesting `if_else` functions would work here, however the `case_when` function would be more appropriate. 
+*Hint: Nesting `if_else` functions would work here - but it would be pretty ugly, The `case_when` function would be more appropriate.*
```

Comment:
@mstackhouse I think the formula syntax could use a quick blurb. What about, use the tilde(~) between the logic and the value in the `case_when` function to create a 'formula' in R.

Reply:
Do you think we need that in the exercise or will it be covered sufficiently in the slides / website?

Reply:
@mstackhouse Good point, I'll add something in the slides instead